### PR TITLE
feat(forms): add support for standalone ngModel dirs inside forms

### PR DESCRIPTION
**Pull request by kara**
This PR allows the use of ngModels within forms without necessarily requiring the registration of their form controls with the parent form.

r: @vsavkin 
cc: @StephenFluin:  can you look at the new error message?

**Before**
If you wanted to include an ngModel within a form tag, it had to have a `name` attribute so that the ngModel could be registered with the parent form properly (for aggregate form validation and value).  

``` html
<form>
    <input [(ngModel)]="person.name" name="first">
    <input [(ngModel)]="person.food">      //  always throws error because name is not set
</form>
```

However, in some rare cases, it might be desirable to have an ngModel within a form tag, but not registered with the form.   This setup was not supported and would throw an error.

**After**

``` html
<form #f="ngForm">
    <input [(ngModel)]="person.name" name="first">
    <input [(ngModel)]="person.food" [ngModelOptions]="{standalone: true}"> 
</form>

{{ f.value | json }}       // {first: ""}
```

Now, if for some reason you need to have a standalone ngModel within a form tag, it's still usable without registering it with the parent form by setting the `standalone` property in `ngModelOptions`.  This way, you can explicitly dictate that you don't want the form registration, but users who may forget the name attribute will still get a helpful error.

It's worth noting that ngModels can still, as always, be used outside of a form tag without a name attribute or an ngModelOptions property.

### CODE REVIEW

File: modules/@angular/forms/src/directives/ng_model.ts [hunk 1]
```diff
@@ -93,30 +92,37 @@ export class NgModel extends NgControl implements OnChanges,
                 return composeAsyncValidators(this._asyncValidators);
               }
 
+              get isStandalone(): boolean {
```
Comment (vsavkin) [inline modules/@angular/forms/src/directives/ng_model.ts]: should it public ir provate?

File: modules/@angular/forms/src/directives/ng_model.ts [hunk 2]
```diff
@@ -93,30 +92,37 @@ export class NgModel extends NgControl implements OnChanges,
...
               private _addControl(): void {
-                this._control = this.formDirective ? this.formDirective.addControl(this) :
-                                                     this._addStandaloneControl();
+                this.isStandalone ? this._addStandaloneControl() :
+                                    this.formDirective.addControl(this);
                 this._added = true;
               }
 
-              private _addStandaloneControl(): FormControl {
+              private _addStandaloneControl(): void {
```
Comment (vsavkin) [inline modules/@angular/forms/src/directives/ng_model.ts]: Not really part of this PR, but should not this be called setUpStandaloneControl? What are you adding it to?

File: modules/@angular/forms/src/directives/ng_model.ts (hunk 1 above)
Comment (kara) [inline modules/@angular/forms/src/directives/ng_model.ts]: Probably private; I'll update.

File: modules/@angular/forms/src/directives/ng_model.ts (hunk 2 above)
Comment (kara) [inline modules/@angular/forms/src/directives/ng_model.ts]: It was named that way so it sounded parallel to "addControl".  I think I'll change both of them to be clearer: setUpControl and setUpStandalone.
